crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2307,11 +2307,26 @@ impl<'db> TypeInferenceBuilder<'db> {
             asname: _,
         } = alias;
 
-        // Check if the symbol being imported is a submodule.  This won't get handled by the
-        // `Type::member` call below because it relies on the semantic index's `imported_modules`
-        // set.  The semantic index does not include information about `from...import` statements
-        // because there are two things it cannot determine while only inspecting the content of
-        // the current file:
+        // First try loading the requested attribute from the module.
+        // TODO: Consider loading _both_ the attribute and any submodule and unioning them together
+        // if both exist.
+        if let Symbol::Type(ty, boundness) = module_ty.member(self.db, name) {
+            if boundness == Boundness::PossiblyUnbound {
+                self.diagnostics.add_lint(
+                    &POSSIBLY_UNBOUND_IMPORT,
+                    AnyNodeRef::Alias(alias),
+                    format_args!("Member `{name}` of module `{module_name}` is possibly unbound",),
+                );
```

Comment:
Nit on the TODO comment. If we did this, it would not be unconditional, it would only be in the case where the attribute exists but is possibly unbound. In that specific scenario, we would ideally given the union because we don't know if the attribute exists or not.
```suggestion
        if let Symbol::Type(ty, boundness) = module_ty.member(self.db, name) {
            if boundness == Boundness::PossiblyUnbound {
                // TODO: Consider loading _both_ the attribute and any submodule and unioning
                // them together if the attribute exists but is possibly-unbound.
                self.diagnostics.add_lint(
                    &POSSIBLY_UNBOUND_IMPORT,
                    AnyNodeRef::Alias(alias),
                    format_args!("Member `{name}` of module `{module_name}` is possibly unbound",),
                );
```

Reply:
Done